badcoding.java

```diff
@@ -0,0 +1,19 @@
+package com.example.springxss;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class XSSController {
+
+    @GetMapping("/hello")
+    ResponseEntity<String> hello(@RequestParam(value = "name", defaultValue = "World") String name) {
+        return new ResponseEntity<>("Hello World!" + name, HttpStatus.OK);
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg" width="20"/></sub>&nbsp; Cross-site Scripting (XSS)

Unsanitized input from an HTTP parameter flows into here, where it is used to render an HTML page returned to the user. This may result in a Cross-Site Scripting attack (XSS).

<h6>
<a href="https://cwe.mitre.org/data/definitions/79.html">CWE-79</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 850</a> | Line 16</h6>